### DISCUSSION TITLE
posts/Go/viper_tutorial/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#338] [Announcements]
https://www.liwenzhou.com/posts/Go/viper_tutorial/ 

 李文周的Blog viper 配置管理 go config yaml json etcd Consul toml 配置 热加载

**Comment by LeonQiuM** [comment]
求指导一下，这种列表长度不固定的怎么解析啊
```
- x:
  a:b
- y:
  a:c
- z:
  a:123
```

**Comment by Q1mi** [comment]
> @LeonQiuM
> 求指导一下，这种列表长度不固定的怎么解析啊
> ```
> - x:
>   a:b
> - y:
>   a:c
> - z:
>   a:123
> ```

viper直接取值或者解析到切片里

**Comment by KurQAQ** [comment]
nice

**Comment by Evil-king** [comment]
请问一下  我循环1000次获取配置 报错 Remote Configurations Error: No Files Found  有遇到么 其实不是有意这样去获取的 只是项目中 频繁获取 就报错了  然后我就用循环去测试

**Comment by Q1mi** [comment]
> @Evil-king
> 请问一下  我循环1000次获取配置 报错 Remote Configurations Error: No Files Found  有遇到么 其实不是有意这样去获取的 只是项目中 频繁获取 就报错了  然后我就用循环去测试 

可以贴一下代码片段，没看明白你的问题

**Comment by yunis-yao** [comment]
windows 平台 下执行 1.7.2 那个例子，一旦 修改配置文件后，fmt.Println("夭寿啦~配置文件被人修改啦...") 会在控制台显示两次，为什么会执行两次？

**Comment by HungYann** [comment]
> 请问一下 我循环1000次获取配置 报错 Remote Configurations Error: No Files Found 有遇到么 其实不是有意这样去获取的 只是项目中 频繁获取 就报错了 然后我就用循环去测试

哥们，解决了吗？

**Comment by HungYann** [comment]
远程加载etcd遇到, "Remote Configurations Error: No Files Found"错误。
```go
v.AddRemoteProvider("etcd", endpoint, key);
```

**Comment by jinzaigo** [comment]
viper读取etcd/consul，需要匿名导入 import _ "github.com/spf13/viper/remote"，未导入会导致连接失败；

同时viper不支持apollo的读取，可以go get -u github.com/shima-park/agollo，然后import "github.com/shima-park/agollo/viper-remote"，这个包会重写viper中的RemoteConfig的相关方法，以达到能支持apollo的读取
